### DISCUSSION TITLE
TRANSLATE

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#155] [General]
Since we are in uncharted territory with a unicode-supporting Rexx implementation, I wonder what to do with TRANSLATE, which is used to translate characters - as long as they are 8bits, but also for reordering and other wonders.

Changestr could fill in the void, but I'll leave this here as a placeholder for discussion. We might need a unicode version of TRANSLATE; and alternatively, must take care that TRANSLATE itself keeps working with 8 bit bytes.

**Comment by Peter-Jacob** [comment]
I am extending the question: do we need for C2X and C2D a strictly byte-oriented instruction? This would be similar to STRCHAR but loading just one byte, even if it is a unicode character.

**Comment by Peter-Jacob** [comment]
> Since we are in uncharted territory with a unicode-supporting Rexx implementation, I wonder what to do with TRANSLATE, which is used to translate characters - as long as they are 8bits, but also for reordering and other wonders.
> 
> Changestr could fill in the void, but I'll leave this here as a placeholder for discussion. We might need a unicode version of TRANSLATE; and alternatively, must take care that TRANSLATE itself keeps working with 8 bit bytes.
I wouldn't go for changestr, as it is more a string replacement, rather than a char replacement, besides that it is really expensive using POS and substr. I see the problem in providing the in/out translation tables as we don't know if the character is 2,4,6,8 bytes long. 
maybe we need a TRANSLATEUNI function with a slightly different syntax.
